### DISCUSSION TITLE
Error log: Unknown receipt type in envelope

### DISCUSSION THREAD

**Discussion by moppman** [isamert/scli#192] [Q&A] [answered]
I noticed this type of error in the scli error log:

`ERROR:root:on_receive_receipt: unknown receipt type in envelope {'source': '+xxx', 'sourceNumber': '+xxx', 'sourceUuid': 'xxx', 'sourceName': 'xxx', 'sourceDevice': 1, 'timestamp': xxx, 'receiptMessage': {'when': xxx, 'isDelivery': False, 'isRead': False, 'isViewed': True, 'timestamps': [xxx]}, '_received_timestamp': xxx}`

Looking at the relevant `scli` [code](https://github.com/isamert/scli/blob/11d3a394d7c7f860b400d0e5dbcd01e4793716a9/scli#L1749), there are two types of recognized receipts: `isDelivery` and `isRead`, probably corresponding to the two well-known checkmarks.

So I'm curious: what type of receipt is `signal-cli` receiving that is neither `isDelivery` nor `isRead`?

**Comment by exquo** [comment, marked as answer]
Good question, I'd like to know as well! I've asked upstream: https://github.com/AsamK/signal-cli/discussions/948

**Comment by exquo** [comment]
Now the `isViewed` receipts are ignored by scli rather than log an error (66e1dc30a0ac1ccd9f65f273fcd39b162c7c9967).
